app/gui/view/component-browser/component-group/src/lib.rs

```diff
@@ -72,6 +102,37 @@ pub mod background {
 }
 
 
+// === Header Background ===
+
+/// The background of the header. It consists of a rectangle that matches the [`background`] in
+/// color and a shadow underneath it.
+pub mod header_background {
+    use super::*;
+
+    ensogl_core::define_shape_system! {
+        above = [background, list_view::background];
+        (style:Style, color:Vector4, height: f32, shadow_height_multiplier: f32) {
+            let color = Var::<color::Rgba>::from(color);
+            let width: Var<Pixels> = "input_size.x".into();
+            let height: Var<Pixels> = height.into();
+            let bg = Rect((width.clone(), height.clone())).fill(color);
+            // We use wider and shorter rect for the shadow because of the visual artifacts that
+            // will appear otherwise:
+            // 1. Rounded corners of the shadow are visible if the rect is too narrow. By widening
+            //    it we keep the shadow sharp and flat for the whole width of the header.
+            // 2. Visual glitching similar to z-fighting occurs on the border of the elements
+            //    when the shadow rect has the exact same size as the background. We shrink the
+            //    height by 1 pixel to avoid it.
+            let shadow_rect = Rect((width * 2.0, height - 1.0.px()));
```

Comment:
Are you using negative values here? Seems so. This is not decribed in the docs. The docs tell `We use wider and shorter rect`, but I', not sure if rect of height -1 is shorter than rect of height 0 (theoretically, both do not exist). I would just write explicitly that when the height is set to 0, we draw a rect of negative height in order to hide anti-aliasing artifacts when scrolling. BTW, FYI, the artifacts appear because anti-aliasign works on pixels close to the border. If the height is set to 0, then theoreticaly, pixels around it are close to border. Hmm, writing this explanation just made me realize how to remove all such artifacts automatically from the rasterization algorithm. Damn, I was thinking about it for months long time ago, but never thought about sampling 4 pixels around the current pixel and discarding the current pixels if at least 2 (top,bottom) or (left,right) are outside of the shape. Damn, it's so simple and wise, thanks Ilya! :D

Reply:
No, actually we're not using negative values here. `height` is the height of the header (header background), and it never equals 0. We use `shadow_height_multiplier = 0.0` to hide the shadow, but that just means that the shadow will not extend beyond its basic shape (which is `[(width*2) x (height - 1)]`). If I understand correctly, the artifacts come from the fact that the basic shape of the shadow is the same as the shape of the background, and the pixels on the border sometimes can't decide which shape they belong to.

This is how the basic shape of the shadow looks like with `shadow_height_multiplier = 0.0`:

<img width="212" alt="Screenshot 2022-05-17 at 12 30 26" src="https://user-images.githubusercontent.com/6566674/168779352-da5ba3c4-a8ee-41a9-aa60-a37a0847264b.png">

(it has a small offset down as per default shadow parameters in our stylesheet)

This is how the shadow looks with non-zero `shadow_height_multiplier`:

<img width="207" alt="Screenshot 2022-05-17 at 12 30 46" src="https://user-images.githubusercontent.com/6566674/168779486-6387a92b-9a98-420c-a3a6-af8d22a05df3.png">